### PR TITLE
[scan] support toolchain parameter

### LINKED ISSUE #8167: Scan - module compiled with Swift 3.0 cannot be imported in Swift 3.0.2
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
When running scan, unable to pass the swift toolchain as toolchain is not among the parameters scan has. xcodebuild has the toolchain parameter. Our Gymfile has the below configuration, but I guess scan uses xcodebuild and not gym to build the tests 

Gymfile
workspace "App.xcworkspace"
output_name "App"
configuration "Release"
clean true
scheme "App"
toolchain "org.swift.3020160913a"

### PR DESCRIPTION
**Pull request by hjanuschka**
fixes: https://github.com/fastlane/fastlane/issues/8167

### CODE REVIEW

File: scan/spec/test_command_generator_spec.rb [hunk 1]
```diff
@@ -74,6 +74,38 @@
...
+        expect do
+          Fastlane::FastFile.new.parse("lane :test do
+            scan(
+              project: './scan/examples/standard/app.xcodeproj',
+              xctestrun: './path/1.xctestrun',
+              toolchain: 'com.apple.dt.toolchain.Swift_2_3'
+            )
+          end").runner.execute(:test)
+        end.to raise_error("Unresolved conflict between options: 'toolchain' and 'xctestrun'")
+      end
```
Comment (DanToml) [inline scan/spec/test_command_generator_spec.rb]: there should be a newline here

File: scan/spec/test_command_generator_spec.rb [hunk 2]
```diff
@@ -74,6 +74,38 @@
...
+      it "Supports toolchain" do
```
Comment (DanToml) [inline scan/spec/test_command_generator_spec.rb]: This should be a part of the `:toolchain` `describe`, and should maybe be `passes the toolchain option to xcodebuild`

**Review by endocrimes — CHANGES_REQUESTED** [review]
Couple of test changes, but looking good :+1:

File: scan/spec/test_command_generator_spec.rb (hunk 1 above)
Comment (hjanuschka) [inline scan/spec/test_command_generator_spec.rb]: fixed

**Comment by hjanuschka** [comment]
@DanToml feedback addressed, thx!

**Review by endocrimes — APPROVED** [review]

**Pull request merged**